Notebook on shortest path using Dijkstra's algorithm

NikitaSharma1 opened the pull request:
I created a notebook on Dijkstra's algorithm and how it is implemented in NetworkX. I would like to know if there's any other topic I can add or explain in more detail.

NikitaSharma1 committed:
Dijkstra's notebook

NikitaSharma1 committed:
Dijkstra's notebook

NikitaSharma1 committed:
minor changes

MridulS commented:
Thanks @NikitaSharma1! Can you please convert the `.ipynb` notebook format to `.md`? You can use [`jupytext`](https://jupytext.readthedocs.io/en/latest/) for this.
With `jupytext --to markdown notebook.ipynb` you can convert it to markdown.

[we really need to get the contributing guide for nx-guides up and running 😅 ]

NikitaSharma1 commented:
@MridulS I converted it to md but I am getting `make: *** [Makefile:20: html] Error 1`  this error again and again . Can you please tell me how to resolve this? 
And yes, contribution guide is much needed 😅

MridulS commented on content/algorithms/Dijkstra/Dijkstra.md:
```diff
@@ -0,0 +1,241 @@
+- It is a greedy algorithm that means it is possible for the algorithm to select the current best option which can make the algorithm get sidetracked following a potential path that doesn’t exist, simply because the edges along it form a short path.
+
+
+#### Reference
```
```suggestion
## Reference
```

myst doesn't like jumping from level 2 header to level 4 header, it needs to be either 2 or 3

MridulS commented:
The build passes now :)
https://output.circle-artifacts.com/output/job/6a1b520d-4125-45f7-af94-41f04c4676e0/artifacts/0/site/_build/html/content/algorithms/Dijkstra/Dijkstra.html

MridulS commented on content/algorithms/Dijkstra/Dijkstra.md:
```diff
@@ -0,0 +1,241 @@
+    - ZRH : `min(infinity, 5) = 5`
+
+
+![Figure 1&2](Graphs/figure1_2.png "Step 1 and Step 2")
```
How were these images generated? It's usually a good idea to keep the code handy to reproduce the figures :)

rossbar replied:
Yes, these images should be generated directly by code in the notebook - showing how to make high-quality visualizations of graph/network data is one of the primary goals of these tutorials!

NikitaSharma1 replied:
I used [diagrams.net](https://www.diagrams.net/) for the graphs for 2 reasons:
1. I think inserting code for graphs might break the flow of the learner as it happened to me in Dintz and I have like 6 graphs.
2. If you'll look in the images, I can add edge labels, node labels and colour edges, but, I don't know how to add the distance of the nodes in the graph (like infinite initially and then keep updating).

But, yes you're right. I'll try to figure it out and create an easy and understandable graph using networkx.

NikitaSharma1 commented:
@MridulS thank you so much!!

MridulS commented on content/algorithms/Dijkstra/Dijkstra.md:
````diff
@@ -0,0 +1,241 @@
+
+# printing the distance and path from the source node 'DEL' to target node 'LCY'                    
+print(dist[target], paths[target])
+```
````
BTW we don't need to implement the algorithm in the same exact way as done inside NetworkX for nx-guides. The goal is to provide a pedagogical source so if possible feel free to remove bits that you think can be better compressed :)

NikitaSharma1 replied:
So, can I replace it with a basic general implementation?

MridulS commented on content/algorithms/Dijkstra/Dijkstra.md:
```diff
@@ -0,0 +1,241 @@
+## Reference
+
+Shivani Sanan, Leena jain, Bharti Kappor (2013). (IJAIEM) "Shortest Path Algorithm" <br>
+https://www.ijaiem.org/volume2issue7/IJAIEM-2013-07-23-079.pdf
```
Why is this paper cited in references? This doesn't define the Dijkstra algorithm.
It's defined in:
```
Dijkstra, Edsger W. "A note on two problems in connexion with graphs." Numerische mathematik 1, no. 1 (1959): 269-271.
```
https://ir.cwi.nl/pub/9256/9256D.pdf

MridulS reviewed (commented):
Just some general comments.

NikitaSharma1 commented:
@MridulS @rossbar I have made the changes you asked for, can you please review them?

NikitaSharma1 committed:
merge

NikitaSharma1 committed:
Merge branch 'networkx:main' into main

NikitaSharma1 committed:
myst

NikitaSharma1 committed:
format

NikitaSharma1 commented:
@MridulS I changed the format of the document, added better graphs, and simplified the code.
But now I am again getting `make: *** [Makefile:20: html] Error 1` error. I rechecked my notebook and there isn't any jump from level 2 to level 4. All headers are of level 2 or level 3 only.

NikitaSharma1 committed:
Merge branch 'networkx:main' into main

MridulS committed:
Merge branch 'main' into main
